File: willfix/src/main/java/wtf/s1/willfix/WillFixExtension.java
package wtf.s1.willfix;

import java.util.List;

import wtf.s1.willfix.logging.LevelLog;

public class WillFixExtension {

    /**
     * 是否开启该插件
     */
    public boolean enable = true;

    public boolean isEnable() {
        return enable;
    }

    public void enable(boolean enable) {
        this.enable = enable;
    }

    public List<String> getMethodList() {
        return methodList;
    }

    public void setMethodList(List<String> methodList) {
        this.methodList = methodList;
    }

    public String getSeparator() {
        return separator;
    }

    public void setSeparator(String separator) {
        this.separator = separator;
    }

    /**
     * 需要用TryCatchBlock包住的方法
     */
    public List<String> methodList;

    public String separator = "#";

    private LevelLog.Level level = LevelLog.Level.INFO;

    public void logLevel(String level) {
        this.level = LevelLog.Level.valueOf(level);
    }

    public LevelLog.Level getLogLevel() {
        return level;
    }

}

File: willfix-standalone/src/main/java/wtf/s1/willfix/WillFixExtension.kt
package wtf.s1.willfix

import wtf.s1.willfix.logging.LevelLog

open class WillFixExtension {
    /**
     * 是否开启该插件
     */
    @JvmField
    var isEnable = true
    @JvmField
    var isEnableInDebug = false
    @JvmField
    var isNeedVerify = true

    /**
     * 需要用TryCatchBlock包住的方法
     */
    var methodList: List<String>? = null

    @JvmField
    var separator = "#"

    var exceptionHandler: String? = null

    var logLevel = LevelLog.Level.INFO
        private set

    fun enable(enable: Boolean) {
        isEnable = enable
    }

    fun logLevel(level: String?) {
        logLevel = LevelLog.Level.valueOf(level!!)
    }

    fun enableInDebug(enableInDebug: Boolean) {
        isEnableInDebug = enableInDebug
    }

    fun needVerify(needVerify: Boolean) {
        isNeedVerify = needVerify
    }

    fun exceptionHandler(catchHandler: String?) {
        exceptionHandler = catchHandler
    }
}